src/network-detect.js

```diff
@@ -89,7 +89,7 @@ NetworkDetect.prototype = {
     }
 
     //Check if number contains unwanted characters
-    if (this.phone.match(/[^0-9]/)) {
+    if (this.phone.slice(1, this.phone.length).match(/[^0-9]/) && this.phone[0] !== '+') {
```

Comment:
With this approach, it doesn't consider the **+** as unwanted if it's at the start of the digits. However, if I put a **+** or any other character somewhere other than position 0, this rule will fail (which we don't want).

Like so: +234819847360+, +234819847&608, +234#198473608 and others.

Also, digits like +2348198473 will also pass since they are 11 digits and start with a **+**.

You need to instead check and ensure it contains just one **+** and at position 0 only for 14 digits, this rule should not work for 11 digits.

Reply:
@BolajiAyodeji 
I have noted the corrections. Thanks for the review